velox/common/caching/tests/AsyncDataCacheTest.cpp

```diff
@@ -40,9 +41,17 @@ struct Request {
 class AsyncDataCacheTest : public testing::Test {
  protected:
   static constexpr int32_t kNumFiles = 100;
-  void initializeCache(int64_t maxBytes) {
+
+  void TearDown() override {
+    if (executor_) {
```

Comment:
nit: Why can't this be `executor()->join()`? This would make the code simpler as it won't require a mutex in the body of executor().